main/puzzle/neo/arduino-neopuzzle/arduino-neopuzzle.ino

```diff
@@ -0,0 +1,84 @@
+#include <Wire.h>
+#include <Adafruit_NeoTrellis.h>
+
+#define MATRIX_SIZE 8
+#define INT_PIN 5 // Interrupt pin for the NeoTrellis
+
+enum NeoState {
+    NEO_UNINITIALIZED,
+    NEO_PLAYING,
+    NEO_SOLVED
+};
+
+Adafruit_NeoTrellis trellis;
+NeoState neoState = NEO_UNINITIALIZED;
+
+// Initialize the NeoTrellis matrix
+void initializeNeoMatrix() {
+    if (!trellis.begin()) {
+        Serial.println("Failed to initialize NeoTrellis");
+        while (1); // Hold here if initialization fails
+    }
+
+    // Set all buttons to listen for presses and releases
+    for (int i = 0; i < MATRIX_SIZE; i++) {
+        for (int j = 0; j < MATRIX_SIZE; j++) {
+            trellis.activateKey(i * MATRIX_SIZE + j, SEESAW_KEYPAD_EDGE_RISING, true);
+            trellis.activateKey(i * MATRIX_SIZE + j, SEESAW_KEYPAD_EDGE_FALLING, true);
+            trellis.setPixelColor(i * MATRIX_SIZE + j, 0x000000); // Turn off LED
+        }
+    }
+    trellis.show();
+    neoState = NEO_PLAYING;
+}
+
+// Callback to handle button presses
+void buttonCallback(uint8_t x) {
+    uint8_t i = x / MATRIX_SIZE;
+    uint8_t j = x % MATRIX_SIZE;
+
+    // Toggle the central button and adjacent LEDs
+    toggleAdjacentLEDs(i, j);
+    if (isNeoPuzzleSolved()) {
+        neoState = NEO_SOLVED;
+        Serial.println("The NeoTrellis puzzle is solved!");
+        // Additional actions upon solving the puzzle can go here
+    }
+    trellis.show();
+}
+
+void toggleAdjacentLEDs(int x, int y) {
+    int idx = x * MATRIX_SIZE + y;
+    trellis.setPixelColor(idx, trellis.getPixelColor(idx) ^ 0xFFFFFF); // Toggle LED color
+
+    // Toggle adjacent LEDs
+    if (x > 0) trellis.setPixelColor((x-1) * MATRIX_SIZE + y, trellis.getPixelColor((x-1) * MATRIX_SIZE + y) ^ 0xFFFFFF);
+    if (x < MATRIX_SIZE - 1) trellis.setPixelColor((x+1) * MATRIX_SIZE + y, trellis.getPixelColor((x+1) * MATRIX_SIZE + y) ^ 0xFFFFFF);
+    if (y > 0) trellis.setPixelColor(x * MATRIX_SIZE + (y-1), trellis.getPixelColor(x * MATRIX_SIZE + (y-1)) ^ 0xFFFFFF);
+    if (y < MATRIX_SIZE - 1) trellis.setPixelColor(x * MATRIX_SIZE + (y+1), trellis.getPixelColor(x * MATRIX_SIZE + (y+1)) ^ 0xFFFFFF);
+}
+
+bool isNeoPuzzleSolved() {
+    for (int i = 0; i < MATRIX_SIZE; i++) {
+        for (int j = 0; j < MATRIX_SIZE; j++) {
+            if (trellis.getPixelColor(i * MATRIX_SIZE + j) != 0x000000) return false; // If any LED is on, puzzle is not solved
```

Comment:
De kleur van knoppen gebruiken vind ik (persoonlijk) een beetje stinky, en gaat waarschijnlijk voor verwarring zorgen als er ooit iets veranderd moet worden aan de kleuren van de neotrellis puzzel. Zet op z'n minst de kleuren in `#define` macro's, maar ik heb liever dat de speltoestand-matrix losgekoppeld wordt van de kleuren die momenteel weergeven worden op de knoppenmatrix.

Aan de andere kant denk ik ook dat het \*maar\* een spel van ~80 regels code is, en dat het dan ook niet heel veel uitmaakt, hoe denkt de rest hier over?

Reply:
Ik begrijp dat de kleurcode's noodzakelijk zijn om de leds aan te sturen, maar ik vind het alleen geen mooie oplossing om de spel state compleet afhankelijk te maken van de kleur van de leds door de state alleen daar op te slaan. Ik bedoelde dat er eigenlijk een losse 8x8 array moet komen met booleans om de binaire speltoestand op te slaan, en dat je daarmee de leds loskoppelt van de speltoestand.

Stel iemand wil in de toekomst dat de knoppen niet alleen uit of op wit staan, maar dat als ze aan zijn dat er een kleurovergang over het spelbord gaat, zou deze aanpak uit elkaar vallen. Voor nu zal het geen problemen veroorzaken, dus ik vind het al prima dat je ze nu in `#define`s hebt gezet.